src/etc/inc/util.inc

```diff
@@ -2542,4 +2542,41 @@ function validateipaddr(&$addr, $type, $label, &$err_msg, $alias=false) {
 
 	return false;
 }
+
+/* returns true if $dhcp6duid is a valid duid entrry */
+function is_duid($dhcp6duid) {
+	$values = explode(":", $dhcp6duid);
+	if (count($values) != 16 || strlen($dhcp6duid) != 47) {
+		return false;
+	}
+	for ($i = 0; $i < 16; $i++) {
+		if (ctype_xdigit($values[$i]) == false)
+			return false;
+		if (hexdec($values[$i]) < 0 || hexdec($values[$i]) > 255)
+			return false;
+	}
+	return true;
+}
+
+/* Write the DHCP6 DUID file */
+function write_dhcp6_duid($duidfile) {
+	// Create the hex array from the dhcp6duid config entry and write to file
+	global $g;
+ 	
+ 	if(is_duid($duidfile))
+ 	{
+ 		$temp = str_replace(":","",$duidfile);
+ 		$duid_binstring = pack("H*",$temp);
+ 		if ($fd = fopen("{$g['vardb_path']}/dhcp6c_duid", "wb")) {
+ 			fwrite($fd, $duid_binstring);
+			fclose($fd);
+ 			return;
+ 		}
+ 		else {
```

Comment:
Else is unnecessary, if code flow went to previous if it will return true and never reach it